spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitStreamMessageHandler.java

```diff
@@ -0,0 +1,259 @@
+	 * @param waitForConfirm true to wait.
+	 * @see #setConfirmTimeout(long)
+	 */
+	public void setWaitForConfirm(boolean waitForConfirm) {
+		this.waitForConfirm = waitForConfirm;
+	}
+
+	/**
+	 * Set the confirm timeout.
+	 * @param confirmTimeout the timeout.
+	 * @see #setWaitForConfirm(boolean)
+	 */
+	public void setConfirmTimeout(long confirmTimeout) {
+		this.confirmTimeout = confirmTimeout;
+	}
+
+	/**
+	 * Set a custom {@link AmqpHeaderMapper} for mapping request and reply headers.
+	 * Defaults to {@link DefaultAmqpHeaderMapper#outboundMapper()}.
+	 * @param headerMapper the {@link AmqpHeaderMapper} to use.
+	 */
+	public void setHeaderMapper(AmqpHeaderMapper headerMapper) {
+		Assert.notNull(headerMapper, "headerMapper must not be null");
+		this.headerMapper = headerMapper;
+	}
+
+	/**
+	 * When mapping headers for the outbound message, determine whether the headers are
+	 * mapped before the message is converted, or afterwards. This only affects headers
+	 * that might be added by the message converter. When false, the converter's headers
+	 * win; when true, any headers added by the converter will be overridden (if the
+	 * source message has a header that maps to those headers). You might wish to set this
+	 * to true, for example, when using a
+	 * {@link org.springframework.amqp.support.converter.SimpleMessageConverter} with a
+	 * String payload that contains json; the converter will set the content type to
+	 * {@code text/plain} which can be overridden to {@code application/json} by setting
+	 * the {@link AmqpHeaders#CONTENT_TYPE} message header. Default: false.
+	 * @param headersMappedLast true if headers are mapped after conversion.
+	 */
+	public void setHeadersMappedLast(boolean headersMappedLast) {
+		this.headersMappedLast = headersMappedLast;
+	}
+
+	/**
+	 * Return the {@link RabbitStreamOperations}.
+	 * @return the operations.
+	 */
+	public RabbitStreamOperations getStreamOperations() {
+		return this.streamOperations;
+	}
+
+	@Override
+	protected void handleMessageInternal(Message<?> requestMessage) {
+		ListenableFuture<Boolean> future;
+		com.rabbitmq.stream.Message streamMessage;
+		if (requestMessage.getPayload() instanceof com.rabbitmq.stream.Message) {
+			streamMessage = (com.rabbitmq.stream.Message) requestMessage.getPayload();
+		}
+		else {
+			MessageConverter converter = streamOperations.messageConverter();
+			org.springframework.amqp.core.Message amqpMessage = mapMessage(requestMessage, converter,
+					this.headerMapper, this.headersMappedLast);
+			streamMessage = this.streamOperations.streamMessageConverter().fromMessage(amqpMessage);
+		}
+		future = this.streamOperations.send(streamMessage);
+		handleConfirms(requestMessage, future);
+	}
+
+	private void handleConfirms(Message<?> message, ListenableFuture<Boolean> future) {
+		future.addCallback(bool -> this.successCallback.onSuccess(message),
+				ex -> this.failureCallback.failure(message, ex));
+		if (this.waitForConfirm) {
```

Comment:
In many other similar `MessageHandler` impls we use concept as `sync`. See the same `KafkaProducerMessageHandler` for example.
I wonder if for consistency we should do the similar over here, too. 
Or I'll change it to `sync` after migrating to SI `6.0` 😄